apollo-router/src/plugins/record_replay/replay_tests.rs

```diff
@@ -0,0 +1,43 @@
+use console::style;
+use tower::ServiceExt;
+
+use super::super::replay::Replay;
+use crate::TestHarness;
+
+#[tokio::test]
+async fn replay_recording() {
+    let recording_file = if let Ok(file) = std::env::var("RECORDING_FILE") {
+        file
+    } else {
+        eprintln!("No recording file to replay");
+        return;
+    };
+
+    let replay = Replay::from_file(recording_file).await.unwrap();
+
+    let req = replay.make_client_request().unwrap();
+    let report = replay.report.clone();
+
+    let test_harness = TestHarness::builder()
+        .schema(&replay.supergraph_sdl())
+        .extra_plugin(replay)
+        .build_router()
+        .await
+        .unwrap();
+
+    let mut resp = test_harness.oneshot(req).await.unwrap();
+    while (resp.next_response().await).is_some() {}
+
+    let report = report.lock().unwrap();
+    let has_items = report.len();
+
+    if has_items == 0 {
+        println!("{}", style("Replay matched the recording 🎉").green());
+    } else {
+        println!();
+        for item in report.iter() {
+            item.print();
+            println!();
+        }
+    }
+}
```

Comment:
Is it a real test ? If yes, it's missing some assert I think

Reply:
it's not, it's just a way to replay a recording and print some output. seemed cheaper than making an xtask task or something, but i'm open to ideas

Reply:
You can just ignore that test to not run it on CI 
```
#[ignore]
```

Reply:
if i do that, i can't run it with cargo test at all. in CI it does nothing